protocol/daemons/bridge/client/client.go

```diff
@@ -73,140 +88,59 @@ func Start(
-	if err != nil {
-		return fmt.Errorf("failed to fetch event params: %w", err)
-	}
-	proposeParams, err := queryClient.ProposeParams(ctx, &bridgetypes.QueryProposeParamsRequest{})
-	if err != nil {
-		return fmt.Errorf("failed to fetch propose params: %w", err)
-	}
-	recognizedEventInfo, err := queryClient.RecognizedEventInfo(ctx, &bridgetypes.QueryRecognizedEventInfoRequest{})
-	if err != nil {
-		return fmt.Errorf("failed to fetch recognized event info: %w", err)
-	}
-
-	// Verify Chain ID.
-	chainId, err := ethClient.ChainID(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to fetch chain ID: %w", err)
-	}
-	if chainId.Uint64() != eventParams.Params.EthChainId {
-		return fmt.Errorf(
-			"expected chain ID %d but node has chain ID %d",
-			eventParams.Params.EthChainId,
-			chainId,
-		)
-	}
-
-	// Fetch logs from Ethereum Node.
-	filterQuery := getFilterQuery(
-		eventParams.Params.EthAddress,
-		recognizedEventInfo.Info.EthBlockHeight,
-		recognizedEventInfo.Info.NextId,
-		proposeParams.Params.MaxBridgesPerBlock,
-	)
-	logs, err := ethClient.FilterLogs(ctx, filterQuery)
-	if err != nil {
-		return fmt.Errorf("failed to fetch logs: %w", err)
-	}
-	telemetry.IncrCounter(
-		float32(len(logs)),
-		metrics.BridgeDaemon,
-		metrics.NewEthLogs,
-		metrics.Count,
-	)
-
-	// Parse logs into bridge events.
-	newBridgeEvents := make([]bridgetypes.BridgeEvent, len(logs))
-	for i, log := range logs {
-		newBridgeEvents[i] = libeth.BridgeLogToEvent(log, eventParams.Params.Denom)
-	}
-
-	// Send bridge events to bridge server.
-	if _, err = serviceClient.AddBridgeEvents(ctx, &api.AddBridgeEventsRequest{
-		BridgeEvents: newBridgeEvents,
-	}); err != nil {
-		return fmt.Errorf("failed to add bridge events: %w", err)
-	}
-
-	// Success.
-	return nil
-}
-
-// getFilterQuery returns a query to fetch logs of bridge events with following filters:
-// - logs are emitted by contract at address `contractAddressHex`.
-// - block height is between `fromBlock` and current finalized block height (both inclusive).
-// - event IDs are sequential integers between `firstId` and `firstId + numIds - 1` (both inclusive).
-func getFilterQuery(
-	contractAddressHex string,
-	fromBlock uint64,
-	firstId uint32,
-	numIds uint32,
-) eth.FilterQuery {
-	// Generate `ethcommon.Hash`s of the next `numIds` event IDs.
-	eventIdHashes := make([]ethcommon.Hash, numIds)
-	for i := uint32(0); i < numIds; i++ {
-		eventIdHashes[i] = ethcommon.BigToHash(new(big.Int).SetUint64(uint64(firstId + i)))
-	}
-
-	return eth.FilterQuery{
-		FromBlock: new(big.Int).SetUint64(fromBlock),
-		ToBlock:   big.NewInt(ethrpc.FinalizedBlockNumber.Int64()),
-		Addresses: []ethcommon.Address{ethcommon.HexToAddress(contractAddressHex)},
-		Topics: [][]ethcommon.Hash{
-			{ethcommon.HexToHash(constants.BridgeEventSignature)},
-			eventIdHashes,
-		},
+) {
+	// Run the main task loop at an interval.
+	for {
+		select {
+		case <-ticker.C:
+			if err := s.RunBridgeDaemonTaskLoop(
```

Comment:
To confirm, this loop is sync and blocking right?

For example, if the ticker happens every 3 seconds and this task loop takes 1 second, then does that mean that the `RunBridgeDaemonTaskLoop` will run every 4 seconds? Is that correct or does this task run every 3 seconds somehow?

Reply:
The ticker does drop signals that are unconsumed, so yes, if the ticker channel is not drained, it will not add new ticks.

I can modify this by running the task loop in a go routine. I believe the same limitation applied to the [previous implementation](https://github.com/dydxprotocol/v4-chain/blob/98d817dbf570d5bda7eb2eb5019d0dbebd424576/protocol/daemons/bridge/client/client.go#L83) as well.